test/sql/join/asof/test_asof_join_pushdown.test

```diff
@@ -79,3 +79,46 @@ ORDER BY ALL
 ----
 5	6
 10	11
+
+statement ok
+CREATE TABLE issue13899(seq_no INT, amount DECIMAL(10,2));
+
+statement ok
+INSERT INTO issue13899 VALUES
+	(1,1.00),
+	(2,null),
+	(3,null),
+	(4,null),
+	(5,2.00),
+	(6,null),
+	(7,null),
+	(8,3.00),
+	(9,null),
+	(10,null),
+	(11,5.00);
+
+# Without the pushdown, the filter does not get applied properly
+statement ok
+PRAGMA disable_verification
```

Comment:
We shouldn't require optimizers for correctness - perhaps this needs to be fixed in the planning stage instead of in the optimizer?